docs/classic-ui/views.md

```diff
@@ -1,13 +1,1007 @@
+  which is a wrapper class. It wraps [zope.publisher.browser.BrowserView](https://github.com/zopefoundation/zope.publisher/blob/master/src/zope/publisher/browser.py#L896)
+  and adds an acquisition (parent traversal) support for it.
+- Views have `index` attribute which points to {doc}`TAL page template </adapt-and-extend/theming/templates_css/template_basics>`
+  responsible rendering the HTML code. You get the HTML output by doing self.index() and page template
+  gets a context argument `view` pointing to the view class instance. `index` value
+  is usually instance of [Products.Five.browser.pagetemplate.ViewPageTemplateFile](https://github.com/zopefoundation/Zope/blob/master/src/Products/Five/browser/pagetemplatefile.py#L33)
+  (full Plone pages) or [zope.pagetemplate.pagetemplatefile.PageTemplateFile](https://github.com/zopefoundation/zope.pagetemplate/blob/master/src/zope/pagetemplate/pagetemplatefile.py#L40)
+  (HTML snippets, no acquisition)
+- View classes should implement {doc}`interface </develop/addons/components/interfaces>`
+  [zope.browser.interfaces.IBrowserView](https://github.com/zopefoundation/zope.browser/blob/master/src/zope/browser/interfaces.py#L27)
+
+Views rendering page snippets and parts can be subclasses of zope.publisher.browser.BrowserView directly
+as snippets might not need acquisition support which adds some overhead to the rendering process.
+
+## Helper views
+
+Not all views need to return HTML output, or output at all.
+Views can beused as helpers in the code to provide APIs to objects. Since views
+can be overridden using layers, a view is a natural plug-in point which an
+add-on product can customize or override in a conflict-free manner.
+
+View methods are exposed to page templates and such, so you can also call
+view methods directly from a page template, not only from Python code.
+
+### More information
+
+- {doc}`Context helpers </develop/plone/misc/context>`
+- {doc}`Expressions </develop/plone/functionality/expressions>`
+
+### Historical perspective
+
+Often, the point of using helper views is that you can have reusable
+functionality which can be plugged in as one-line code around the system.
+Helper views also get around the following limitations:
+
+- TAL security.
+- Limiting Python expression to one line.
+- Not being able to import Python modules.
+
+```{Note}
+Using `RestrictedPython` scripts (creating Python through the
+Management Interface) and Zope 2 Extension modules is discouraged.
+The same functionality can be achieved with helper views, with less
+potential pitfalls.
+```
+
+## Reusing view template snippets or embedding another view
+
+To use the same template code several times you can either:
+
+- create a separate `BrowserView` for it and then call this view (see
+  [Accessing a view instance in code] below);
+- share a `ViewPageTemplate` instance between views and using it several
+  times.
+
+```{Note}
+The Plone 2.x way of doing this with TAL template language macros is
```

Comment:
"Plone 2.x"? Is this note even relevant for Plone 6? Can this note be deleted?

Reply:
Since we marcos are still part of the stack and being used in core, the statement should stay.
But it needs rephrasing.